v21.1/delete-data.md

```diff
@@ -78,38 +122,99 @@ try (Connection connection = ds.getConnection()) {
+
+{{site.data.alerts.callout_success}}
+If the `WHERE` clause evaluates to `TRUE` for a large number of rows (i.e., tens of thousands of rows), use a [batch-delete loop](bulk-delete-data.html#batch-delete-on-an-indexed-column) instead of executing a simple `DELETE` query.
+{{site.data.alerts.end}}
+
+#### Delete rows filtered on a unique column
+
+Suppose that you want to delete the promo code data for a specific set of codes. To delete the rows in the `promo_codes` table where the `code` matches a string in a set of string values:
+
+<div class="filters clearfix">
+  <button class="filter-button" data-scope="sql">SQL</button>
+  <button class="filter-button" data-scope="go">Go (lib/pq)</button>
+  <button class="filter-button" data-scope="java">Java (jdbc)</button>
+  <button class="filter-button" data-scope="python">Python (psycopg2)</button>
+</div>
+
+<section class="filter-content" markdown="1" data-scope="sql">
+
+{% include copy-clipboard.html %}
+~~~ sql
+DELETE from promo_codes WHERE code IN ('0_explain_theory_something', '100_address_garden_certain', '1000_do_write_words');
+~~~
+
+For more information about how to use the built-in SQL client, see the [`cockroach sql`](cockroach-sql.html) reference docs.
 
 </section>
 
-## Delete multiple rows
+<section class="filter-content" markdown="1" data-scope="go">
 
-You can delete multiple rows from a table in several ways:
+{% include copy-clipboard.html %}
+~~~ go
+// tx is a *sql.Tx from "database/sql"
+
+codeOne := "0_explain_theory_something"
+codeTwo := "100_address_garden_certain"
+codeThree := "1000_do_write_words"
+
+if _, err := tx.Exec("DELETE from promo_codes WHERE code IN ($1, $2, $3)", codeOne, codeTwo, codeThree); err != nil {
+  return err
+}
+return nil
+~~~
 
-- Using a `WHERE` clause to limit the number of rows based on one or more predicates:
+</section>
 
-    {% include copy-clipboard.html %}
-    ~~~ sql
-    DELETE FROM student_loan_accounts WHERE loan_amount < 30000;
-    ~~~
+<section class="filter-content" markdown="1" data-scope="java">
 
-- Using a `WHERE` clause to specify multiple records by a specific column's value (in this case, `id`):
+{% include copy-clipboard.html %}
+~~~ java
+// ds is an org.postgresql.ds.PGSimpleDataSource
 
-    {% include copy-clipboard.html %}
-    ~~~ sql
-    DELETE FROM accounts WHERE id IN (1, 2, 3, 4, 5);
-    ~~~
+String codeOne = "0_explain_theory_something";
+String codeTwo = "100_address_garden_certain";
+String codeThree = "1000_do_write_words";
 
-- Using [`TRUNCATE`](truncate.html) instead of [`DELETE`](delete.html) to delete all of the rows from a table, as recommended in our [performance best practices](performance-best-practices-overview.html#use-truncate-instead-of-delete-to-delete-all-rows-in-a-table).
+try (Connection connection = ds.getConnection()) {
+    connection.createStatement().executeUpdate(
+            "DELETE from promo_codes WHERE code IN('" + codeOne + "','" + codeTwo + "','" + codeThree + "')");
```

Comment:
it's risky to put in raw strings like this into SQL statements, since it allows for SQL injection. of course, it doesn't matter for this toy example, but we might as well use a best practice here so the reader doesn't copy+paste something dangerous.

```
PreparedStatement p = connection.prepareStatement("DELETE from promo_codes WHERE code IN(?, ?, ?)");
p.setString(1, codeOne);
p.setString(2, codeTwo);
p.setString(3, codeThree);
p.executeUpdate(); 
```

(it also seems unrealistic to write real JDBC code that way, since a framework on top would take care of this, but at least this way is "safe")

Reply:
this is my personal preference, and also would be a big project, but i wonder if we could think about using [JDBI](https://jdbi.org/) for stuff like this -- where we want to run raw SQL with Java. it's a lightweight library that helps with query building and stuff. wouldn't say this is a big priority though.